src/lib.rs

```diff
@@ -597,46 +594,110 @@ impl ConfiguredOidc {
+        headers.push(("Set-Cookie", nonce_cookie_value));
+
         // Build URL
         let url = Url::parse_with_params(
             self.open_id_config.auth_endpoint.as_str(),
             &[
-                ("response_type", "code"),
-                ("code_challenge", &pkce_challenge),
-                ("code_challenge_method", "S256"),
-                ("client_id", &self.plugin_config.client_id),
-                ("redirect_uri",&self.plugin_config.redirect_uri.as_str()),
-                ("scope", &self.plugin_config.scope),
-                ("claims", &self.plugin_config.claims),
-            ],
-        )
-        .unwrap();
+                    ("response_type", "code"),
+                    ("code_challenge", &pkce_challenge),
+                    ("code_challenge_method", "S256"),
+                    ("state", &state_string),
+                    ("client_id", &self.plugin_config.client_id),
+                    ("redirect_uri",&self.plugin_config.redirect_uri.as_str()),
+                    ("scope", &self.plugin_config.scope),
+                    ("claims", &self.plugin_config.claims),
+                ],
+            )
+            .unwrap();
+
+        // Add location header
+        headers.push(("Location", url.as_str()));
 
         // Send HTTP response
         self.send_http_response(
             307,
-            vec![
-                // Original path to redirect back to
-                ("Set-Cookie", &format!("original-path={}; Max-Age={}", &original_path_encoded, 180)),
-                // Set the pkce challenge as a cookie to verify the callback.
-                ("Set-Cookie", &format!("code-verifier={}; Max-Age={}", &pkce_encoded, 180)),
-                // Redirect to `authorization endpoint`
-                ("Location", url.as_str()),
-                ],
-                Some(b"Redirecting..."),
+            // Redirect to `authorization endpoint` along with the cookie
+            headers.to_vec(),
+            Some(b"Redirecting..."),
             );
             return Action::Pause;
     }
+
+    /// Helper function to get the session cookie as a string by getting the cookie from the request
+    /// headers and concatenating all cookie parts.
+    pub fn get_session_cookie_as_string(&self) -> String {
+
+        // Find all cookies that have the cookie_name, split them by ; and remove the name from the cookie
+        // as well as the leading =. Then join the cookie values together again.
+        let cookie = self.get_http_request_header("cookie").unwrap_or_default();
+
+        // Split cookie by ; and filter for the cookie name.
+        let cookie = cookie.split(";")
+            .filter(|x| x.contains(self.plugin_config.cookie_name.as_str()))
+            .filter(|x| !x.contains(format!("{}-nonce", self.plugin_config.cookie_name).as_str()))
+            // Then split by = and get the second element.
+            .map(|x| x.split("=")
+            .collect::<Vec<&str>>()[1])
```

Comment:
Is this safe against manipulated headers?

Reply:
Do you have an example?

Reply:
```
> curl localhost:10000 -H "Cookie: oidcSession"

wasm-oidc-plugin: panicked at 'index out of bounds: the len is 1 but the index is 1'
```